src/main.rs

```diff
@@ -2,11 +2,48 @@ extern crate getopts;
 
 extern crate otp_cop;
 
-use std::{env};
+use std::{env, thread};
+use std::sync::{mpsc};
 
 use otp_cop::service::{CreateServiceResult, ServiceFactory};
 
 
+struct ParallelIter<T> {
+    count: usize,
+    pos: usize,
+    rx: mpsc::Receiver<T>
+}
+
+impl<T> Iterator for ParallelIter<T> {
+    type Item = T;
+
+    fn next(&mut self) -> Option<T> {
+        if self.pos < self.count {
+            self.count += 1;
+            return self.rx.recv().ok();
+        } else {
+            return None;
+        }
+    }
+}
+
+fn parallel<T, U, F1>(objs: Vec<T>, f1: F1) -> ParallelIter<U>
+        where F1: 'static + Fn(T) -> U + Send, T: Send, U: Send {
+    let (tx, rx) = mpsc::channel();
+    let mut threads = vec![];
+    for o in objs {
+        threads.push(thread::spawn(move || {
+            tx.send(f1(o));
+        }));
+    }
+
+    for t in threads {
```

Comment:
I feel like ideally you don't need to join all the threads here – there's no reason the client can't consume as work comes in. You can presumably tie the lifetimes to the lifetime of the `ParallelIter` to make sure things get cleaned up, somehow.